frame/transaction-payment/rpc/src/lib.rs

```diff
@@ -101,4 +137,47 @@ where
 			data: Some(format!("{:?}", e).into()),
 		})
 	}
+
+	fn query_fee_details(
+		&self,
+		encoded_xt: Bytes,
+		at: Option<<Block as BlockT>::Hash>
+	) -> Result<RpcFeeDetails> {
+		let api = self.client.runtime_api();
+		let at = BlockId::hash(at.unwrap_or_else(||
+			// If the block hash is not supplied assume the best block.
+			self.client.info().best_hash
+		));
+
+		let encoded_len = encoded_xt.len() as u32;
+
+		let uxt: Block::Extrinsic = Decode::decode(&mut &*encoded_xt).map_err(|e| RpcError {
+			code: ErrorCode::ServerError(Error::DecodeError.into()),
+			message: "Unable to query fee details.".into(),
+			data: Some(format!("{:?}", e).into()),
+		})?;
+		let fee_details = api.query_fee_details(&at, uxt, encoded_len).map_err(|e| RpcError {
+			code: ErrorCode::ServerError(Error::RuntimeError.into()),
+			message: "Unable to query fee details.".into(),
+			data: Some(format!("{:?}", e).into()),
+		})?;
+
+		let try_into_rpc_balance = |value: Balance| value.try_into().map_err(|_| RpcError {
+			code: ErrorCode::InvalidParams,
+			message: format!("{} doesn't fit in NumberOrHex representation", value),
+			data: None,
+		});
+
+		Ok(RpcFeeDetails {
```

Comment:
Why do we need a new type here? Can we not just use `FeeDetails<NumberOrHex>` here? Than we don't need to copy this type.

Reply:
Done.